src/rosdep2/sources_list.py

```diff
@@ -487,12 +487,21 @@ def update_sources_list(sources_list_dir=None, sources_cache_dir=None,
     python_versions = {}
 
     print('Query rosdistro index %s' % get_index_url())
-    for dist_name in sorted(get_index().distributions.keys()):
+    distribution_names = get_index().distributions.keys()
+    if ros_distro is not None and ros_distro not in distribution_names:
+        raise ValueError(
+            'Requested distribution "%s" is not in the index.' % ros_distro)
+
+    for dist_name in sorted(distribution_names):
         distribution = get_index().distributions[dist_name]
-        if skip_eol_distros:
-            if distribution.get('distribution_status') == 'end-of-life':
-                print('Skip end-of-life distro "%s"' % dist_name)
+        if dist_name != ros_distro:
+            if ros_distro is not None:
```

Comment:
This logic (the two lines here in combination with the for loop) is a bit tricky, but I think it's correct.